### DISCUSSION TITLE
Cases Escalation

### DISCUSSION THREAD

**Discussion by Mav1814** [Security-Onion-Solutions/securityonion#8781] [Unsupported Versions] [answered]
Hi, 
I was wondering if is possible to escalate the alerts to another solution, that's not the Case management?
I'm trying to figure out where escalation button calls are made to try to adapt escalation to another solution.
I've been looking through the documentation, but I haven't found anything concrete.

Cheers to all! And thank you!
mav1814

**Comment by dougburks** [comment, marked as answer]
Please see https://docs.securityonion.net/en/2.3/soc-customization.html#escalation.

**Reply by Mav1814** [reply]
I've been looking through the documentation but didn't help to figure out where escalation button calls are made to try to adapt escalation to another solution.

**Reply by dougburks** [reply]
Did you read through https://docs.securityonion.net/en/2.3/soc-customization.html#escalation? It explains that SOC supports escalating to Cases, TheHive, Elastic Cases, and httpcase. The `httpcase` option can be adapted to another solution.

**Reply by Mav1814** [reply]
yes sir, i did read through the documentation more than once, but i was not able to adapt to another solution.
I've been having trouble understanding the documentation as it's unclear, especially for those trying to understand how security onion works. I apologize for my ignorance.

**Reply by dougburks** [reply]
What exactly are you having trouble understanding?

**Reply by rosswakelin** [reply]
Hey Doug - the httpcase stuff makes sense given the example in the docs, but what is the list of content that can go int the createBody line.  The example has things like .Id and .Title, but what other variables are available?

Ross

**Reply by Mav1814** [reply]
hi,

> What exactly are you having trouble understanding?

For example "To do so, locate the soc [Salt](https://docs.securityonion.net/en/2.3/salt.html#salt) pillar and then set case_module to one of the following values:". I don't know if is clear for everyone, I've been using/learning SO for some time, and I've trouble to locate some configurations files.

**Reply by Mav1814** [reply]
This is also unclear "To use this module, you will need to add a second pillar value, for the pillar httpcase_config".

**Reply by dougburks** [reply]
@Mav1814 

> For example "To do so, locate the soc [Salt](https://docs.securityonion.net/en/2.3/salt.html#salt) pillar and then set case_module to one of the following values:". I don't know if is clear for everyone, I've been using/learning SO for some time, and I've trouble to locate some configurations files.

From the linked Salt documentation at https://docs.securityonion.net/en/2.3/salt.html#salt:
_Minion pillar file: This is the minion specific pillar file that contains pillar definitions for that node. Any definitions made here will override anything defined in other pillar files, including global. This is located at /opt/so/saltstack/local/pillar/minions/<minionid>.sls._

So if you go to the `/opt/so/saltstack/local/pillar/minions/` directory, you should see a file that starts with the hostname of your manager and ends in `_manager.sls`. You should find the `soc` pillar in that file.

**Reply by Mav1814** [reply]
Hi, thank you for the response @dougburks ! 
I Found the soc pillar in the file!
trying to make the escalation work, but is not easy.

**Reply by dougburks** [reply]
There is a full example at the bottom of https://docs.securityonion.net/en/2.3/soc-customization.html#escalation:

> Example of a customized SOC pillar file located in /opt/so/saltstack/local/pillar/minions/import_import.sls (your file path will vary depending on your installation choices):
```
soc:
  es_index_patterns: '*:so-*,*:endgame-*'
  case_module: httpcase
  httpcase_config: |
    "hostUrl": "http://172.17.0.1/some/api",
    "headers": [
      "Authorization: basic Fa3Fa01mDmCC09dA",
      "x-some-key: 1122"
    ],
    "verifyCert": true,
    "createPath": "/some/url/path/to/create/a/case",
    "createMethod": "PUT",
    "createBody": "{\"myid\":\"{{ '{{ .Id }}' }}\", \"title\":\"{{ '{{ .Title }}' }}\", \"desc\":\"{{ '{{ .Description | js }}' }}\"}",
    "createContentType": "application/json",
    "createSuccessCode": 200
```

**Reply by Mav1814** [reply]
Hi @dougburks,

When i try to escalate i get an 400 bad request error from SO.  I cant find any info on the logs.

**Reply by dougburks** [reply]
Are you adjusting the parameters in the example like `hostUrl`, `Authorization`, and `createPath` to match whatever you're sending to?

Have you looked at the SOC logs in `/opt/so/log/soc/` for additional clues?

**Reply by Mav1814** [reply]
yes, i have a question, 
when a make the escalation it  should make the post to hosturl  api right? 
Yes i looked into SOC logs, even clicking on the escalation at the same time. 

MYCONFIG:

soc:
 es_index_patterns: '*:so-*,*:endgame-*'
 case_module: httpcase
  httpcase_config: |
    "hostUrl": "http://192.168.122.212:9000/api",
    "headers": [
      "Authorization: Bearer /uDP5fBVTMcuYOXHpnmegFTqkPPGitHu"
    ],
    "verifyCert": true,
    "createPath": "/case",
    "createMethod": "POST",
    "createBody": "'{\"title\": \"My first case\",\"description\": \"This case has been created by my custom script\",\"severity\": 3}'",
    "createContentType": "application/json",
    "createSuccessCode": 200

**Reply by Mav1814** [reply]
I found something  by digging into the logs. 


`
timestamp=2022-09-29T13:25:08.315603558Z level=warn message="Failed to submit request" error="Post \"http://192.168.122.212:9000/api/case\": dial tcp 192.168.122.212:9000: connect: no route to host"`

and 

`timestamp=2022-09-29T16:28:14.825926674Z level=info message="HTTP request finished" contentLength=266 method=POST status="400 Bad Request" statusCode=400 url=http://192.168.122.212:9000/api/case
`

UPDATE:

Both errors have stopped happening. 

-  The first error happened because the configuration made in /opt/so/saltstack/local/pillar/minions/machinename.sls, was not in the file /root/installtmp/pillar/minions/machinename.sls. After putting the same configuration it stopped. 
- The second error, I believe has to do with the "createBody". Still, although this error does not appear, the 400 error when i try to  escalate still appears.
- I don't have any log now, about the error.

**Reply by Mav1814** [reply]
UPDATE:

When I try to escalate a case and through burpSuite I analyze the requests, I notice that the address I put in the config file (As I posted above the configuration I am using is the same. ) where I want to escalate the case does not show up,  it always tries to escalate inside the URL if SO and not the outside URL.

Changing the IP in burpSuite to the outside IP gives the error 307 Temporary Redirect.

I dont understand what i'm doing wrong.

**Reply by Mav1814** [reply]
Anyone with the same problem??
